data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import org.opensearch.dataprepper.plugins.processor.configuration.KeysConfig;
+import org.opensearch.dataprepper.plugins.processor.configuration.ServiceTypeOptions;
+import org.opensearch.dataprepper.plugins.processor.configuration.AwsAuthenticationOptions;
+
+import java.util.List;
+
+/**
+ * An implementation class of GeoIP Processor configuration
+ */
+public class GeoIPProcessorConfig {
+
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+    @JsonProperty("keys")
+    @NotNull
+    private List<KeysConfig> keysConfig;
```

Comment:
Can this just be List\<KeyConfig\>?

Reply:
Since the structure of YAML configuration has to be changed based on the config PR #2811 This will be addressed as incremental changes later.